src/basic/time-util.h

```diff
@@ -199,3 +199,8 @@ int time_change_fd(void);
 
 const char* timestamp_style_to_string(TimestampStyle t) _const_;
 TimestampStyle timestamp_style_from_string(const char *s) _pure_;
+
+static inline  uint32_t time_to_uint32(__time_t timestamp){
```

Comment:
`__time_t` is an internal type of glibc, we shouldn't use that. why not use regular time_t here?

on systems where time_t is 32bit this won't work.

also, why define this as inline func in the header? only timesyncd needs this, hence move it there.